plugins/inputs/haproxy/haproxy_test.go

```diff
@@ -219,7 +220,32 @@ func TestHaproxyKeepFieldNames(t *testing.T) {
 	fields["hrsp_other"] = fields["http_response.other"]
 	delete(fields, "http_response.other")
 
-	acc.AssertContainsTaggedFields(t, "haproxy", fields, tags)
+	assertContainsMatchingValue(t, &acc, "haproxy", fields, tags)
+}
+
+func assertContainsMatchingValue(
```

Comment:
@danielnelson I'm going to be submitting a similar PR to the `varnish` plugin to add counters, and I'm not really a huge fan of this assert function that I've created. Any ideas on how this can be improved as I couldn't use the existing `acc.AssertContainsTaggedFields()` function as the p.Fields are split across untyped/counter metrics.